scripts/skills/perks/perk_rf_vigorous_assault.nut

```diff
@@ -94,6 +94,49 @@ this.perk_rf_vigorous_assault <- ::inherit("scripts/skills/skill", {
 			}
 		}
 
+		// Reduce the AP cost preemptively by assuming a movement of this.m.BonusEveryXTiles for AI to allow for
+		// AI behaviors which check for AP cost of attacks for setting tile scores when calculating where to go
+		if (!actor.isPlayerControlled() && this.m.NumTilesMoved < this.m.BonusEveryXTiles && actor.isPlacedOnMap())
+		{
+			local aoo = this.getContainer().getAttackOfOpportunity();
+			if (aoo != null)
+			{
+				local numEnemiesInRange = 0;
+				local numEnemiesApproachable = 0;
+				local myTile = actor.getTile();
+
+				foreach (faction in ::Tactical.getAllInstances())
```

Comment:
This needs to be:
`foreach (faction, entityArray in ::Tactical.getAllInstances())`
as getAllInstances returns an array where the indices are equal to the faction IDs

further down must then be `foreach (enemy in entityArray)` or which ever variable name you choose

Reply:
Good catch, thanks.